enhancements/deprecated-bundles.md

```diff
@@ -0,0 +1,131 @@
+- The index add command is enhanced to un-do this termination if needed
+- The deprecation is specified after promotion (not during) and before publishing of the index image
+- Alerts are fired on cluster when a deprecated version is installed
+
+### Non-Goals
+
+- OLM supports upgrading from a terminated bundle
+- Alerts are fired if terminated bundles exist on cluster
+- OLM prevents cluster upgrades based on whether versions are terminated in the upgraded cluster’s index
+
+## Proposal
+
+We propose to add a command to `opm index` called `deprecate` which takes in a set of bundle images, an index, and all the same arguments as `opm index add`. The command modifies the index by removing all versions that come before the specified bundle image in the update graph (that is, until the tail of each channel). If this operation removes the head of a channel, then the channel is removed. The removal of bundles from the update graph will not modify the update graph after the deprecated version (from the deprecated channel to the head of the channel).
+
+Versions that were removed from the index will not receive updates if subscriptions still exist for them on cluster. Ideally, appropriate communication from the operator maintainer will mean that cluster admins will upgrade to the latest supported version prior to cluster upgrade. In the event that the cluster is upgraded and `Subscriptions` to terminated bundles exist on cluster, the cluster admin can manually trigger an upgrade by either:
+
+- editing the `Subscription`'s `channel` field to start receiving updates from a different channel
+- creating a custom `CatalogSource` pointing to the previous OCP index, editing the `Subscription`'s `source` field to point to the new `CatalogSource` in order to receive updates.
+
+With the new work on the skip range being steppable, the above scenario may be less likely to happen.
+
+A new field will be added to the operator-registry database table `operatorbundle` called `deprecated` and the bundle images specified in the command will be marked as such.
+
+The OLM resolver will be modified with a new property type called `olm.deprecated` in order to filter out any options that have this property set and disallow installs of deprecated bundles. This also means appropriate status reporting in the subscription and installplan need to be added to OLM, and alerts will be fired if a deprecated version is installed.
+
+### User Stories
+
+#### Story 1
+
+As an index maintainer I want to be able to truncate update graphs in order to remove or prevent installs of incompatible or unsupported versions of certain operators.
+
+##### Implementation Details
+
+A new field is added to the operator-registry database table `operatorbundle` called `deprecated` and the bundle images specified in the command will be marked as such. Versions that come after the deprecated version in the update graph (in all channels) are removed from the database.
```

Comment:
That come after, tailwards? Meaning, those versions which would get replaced by the deprecated version?

Reply:
yes - I'll make that more clear